e2e/tests/plugins/imagery/exampleImagery.e2e.spec.js

```diff
@@ -455,15 +435,243 @@ test.describe('Example imagery thumbnails resize in display layouts', () => {
+
+        // Zoom in on next image
+        await bgImageLocator.hover();
+        await page.mouse.wheel(0, deltaYStep * 2);
+
+        // Wait for zoom animation to finish
+        await bgImageLocator.hover();
+        const imageNextMouseZoomedIn = await page.locator(backgroundImageSelector).boundingBox();
+        expect(imageNextMouseZoomedIn.height).toBeGreaterThan(originalImageDimensions.height);
+        expect(imageNextMouseZoomedIn.width).toBeGreaterThan(originalImageDimensions.width);
+
+        // Click previous image button
+        await previousImageButton.click();
+
+        // Verify previous image
+        await expect(selectedImage).toBeVisible();
+
+        const imageCount = await page.locator('.c-imagery__thumb').count();
+        await expect.poll(async () => {
+            const newImageCount = await page.locator('.c-imagery__thumb').count();
+
+            return newImageCount;
+        }, {
+            message: "verify that new images still stream in",
+            timeout: 6 * 1000
+        }).toBeGreaterThan(imageCount);
+
+        // Verify selected image is still displayed
+        await expect(selectedImage).toBeVisible();
+
+        // Unpause imagery
+        await page.locator('.pause-play').click();
+
+        //Get background-image url from background-image css prop
+        await assertBackgroundImageUrlFromBackgroundCss(page);
+    });
 });
 
+/**
+ * @param {import('@playwright/test').Page} page
+ */
+async function saveTemplate(page) {
+    await page.locator('.c-button--menu.c-button--major.icon-save').click();
+    await page.locator('text=Save and Finish Editing').click();
+}
+
+/**
+ * @param {import('@playwright/test').Page} page
+ */
+async function assertBackgroundImageUrlFromBackgroundCss(page) {
+    const backgroundImage = page.locator('.c-imagery__main-image__background-image');
+    let backgroundImageUrl = await backgroundImage.evaluate((el) => {
+        return window.getComputedStyle(el).getPropertyValue('background-image').match(/url\(([^)]+)\)/)[1];
+    });
+    let backgroundImageUrl1 = backgroundImageUrl.slice(1, -1); //forgive me, padre
+    console.log('backgroundImageUrl1 ' + backgroundImageUrl1);
+
+    let backgroundImageUrl2;
+    await expect.poll(async () => {
+        // Verify next image has updated
+        let backgroundImageUrlNext = await backgroundImage.evaluate((el) => {
+            return window.getComputedStyle(el).getPropertyValue('background-image').match(/url\(([^)]+)\)/)[1];
+        });
+        backgroundImageUrl2 = backgroundImageUrlNext.slice(1, -1); //forgive me, padre
+
+        return backgroundImageUrl2;
+    }, {
+        message: "verify next image has updated",
+        timeout: 6 * 1000
+    }).not.toBe(backgroundImageUrl1);
+    console.log('backgroundImageUrl2 ' + backgroundImageUrl2);
+}
+
+/**
+ * @param {import('@playwright/test').Page} page
+ */
+ async function panRight(page) {
```

Comment:
@unlikelyzero This feels messy. I have to add all the const's from all over the Test file into this function and will have to do the same thing for the remaining three functions for paning.  Is this the right way to do this?

Reply:
@afahey87 in this particular example, I might move all the pan and zoom functionality and expects into this block and call the function:
`panZoomAndAssertImageProperties`

You might also return values in the function when called to be used in the tests.

Reply:
Okay good idea. Ill try that tonight.

Reply:
It freaking works! 😄

Reply:
@unlikelyzero How about now?

Reply:
@unlikelyzero Ready for review again.